### PR TITLE
Update Commons IO Version and Jackson Validation Tests

### PR DESCRIPTION
**Pull request by alzimmermsft**
Updates `commons-io` to the latest released version and changes the versions used in Jackson validation testing.

### CODE REVIEW

File: sdk/core/supported-version-matrix.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
...
     "TestGoals": "surefire:test",
     "AZURE_CORE_TEST_SUPPORTED_JACKSON_VERSION": [
-        "2.10.5",
-        "2.11.4",
-        "2.12.5",
-        "2.13.2"
+        "2.10.0",
+        "2.11.0",
+        "2.12.0",
+        "2.13.0"
```
Comment (srnagar) [inline sdk/core/supported-version-matrix.json:19]: Why did we downgrade the versions here?

Comment (alzimmermsft) [reply sdk/core/supported-version-matrix.json:19]: My thoughts are we should be validating supported versions against the lowest patch version in the minor version to ensure maximum compatibility

**Review by srnagar — APPROVED** [review]
LGTM

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**